frontend/pages/SoftwarePage/SoftwareTitleDetailsPage/DeleteSoftwareModal/DeleteSoftwareModal.tsx

```diff
@@ -18,13 +18,15 @@ const DELETE_SW_INSTALLED_DURING_SETUP_ERROR_MSG =
 interface IDeleteSoftwareModalProps {
   softwareId: number;
   teamId: number;
+  software: any; // TODO
```

Comment:
Can we set this to `ISoftwarePackage`?

Reply:
Or just pass in the name as a string?

Reply:
Would have to tweak upstream because according to the type checker we might (but in reality won't) have a VPP app instead, which would result in an undefined installer property, so I can't tighten the type up without other changes (type checker shows `ISoftwarePackage | undefined`).

The Edit Software modal has the same issue, so probably makes sense to hit both with the same fix.

Reply:
The way usually handle that is by setting the type as a union with `undefined`, then use optional chaining to handle the undefined case (which we know won't happen) when referencing it. You could also nullish coallesce to a default case for comfort, though since we know it will always be defined it should be okay either way, of course. So something like

`software: ISoftwarePackage | undefined;`
...

`software?.name ?? "this software"`

I'll sometimes leave a comment to the effect of "This will actually always be defined" just to be clear, too.

Reply:
We have quite a few instances of this situation, which is of course not ideal, but very often cleaning up the typing upstream would involve a) untangling a huge mess of spaghetti and/or b) refactoring our API service layer with generics (something we've discussed) to have actual type checking there. Both would be great to do, but of course, require time, effort, and prioritization.

Reply:
@jacobshandling Revised to pass the (nullable) name of the package in [b289060](https://github.com/fleetdm/fleet/pull/23427/commits/b289060ee5dd6a3ddb8f176f5bc74014af36dbe6)